aten/src/ATen/core/type.cpp

```diff
@@ -655,4 +655,90 @@ void checkNoAny(const Type& base, const char* what, const std::string& attrname,
       "' but it contains an Any type. Any types cannot be members of modules, classes, or named tuples.");
 }
 
+namespace {
+bool isSpecialChar(char a) {
+  return a == '[' || a == ',' || a == ']';
+}
+
+class TypeParser {
+ public:
+  explicit TypeParser(const std::string& pythonStr)
+    : pythonStr_(pythonStr), start_(0) {}
+    
+  TypePtr parse() {
+    std::string token;
+    for (; start_ < pythonStr_.size() && !isSpecialChar(pythonStr_[start_]);
+           ++start_) {
+      if (pythonStr_[start_] != ' ') {
+        token += pythonStr_[start_];
+      }
+    }
+    if (token == "int") {
+      return IntType::create();
+    } else if (token == "float") {
+      return FloatType::create();
+    } else if (token == "bool") {
+      return BoolType::create();
+    } else if (token == "str") {
+      return StringType::create();
+    } else if (token == "Any") {
+      return AnyType::create();
+    } else if (token == "Tensor") {
+      return TensorType::create(
```

Comment:
This should use `TensorType::get()` instead, which sets the more obscure flags that come at the end of this constructor correctly for default tensors.

Reply:
Merged to the look up table, where get() is used.